app/controllers/admin/payments_controller.rb

```diff
@@ -0,0 +1,63 @@
+module Admin
+  class PaymentsController < AdminController
+    include PromosHelper
+    include PublishersHelper
+    def show
+      respond_to do |format|
+        format.html
+        format.json do
+          render json: show_data(params[:id])
+        end
+      end
+    end
+
+    def show_data(id)
```

Comment:
This is a massive controller action, I'd probably prefer if we created a model to kind of house all these. In the past I kind of took the idea of "View models" when it came to React front ends. Basically using a model to house and keep this data really clean. And then in the controller just calling `to_json` on the model to serialize it. 

Perhaps in the future we can apply a similar pattern.

Reply:
I think this is a great idea, could you tell me more? What should we call the View Model, and what's the best way to pass a publisher id to it.